pystac/extensions/scientific.py

```diff
@@ -176,23 +205,33 @@ def remove_publication(self, publication: Optional[Publication] = None) -> None:
                 for one_pub in pubs:
                     remove_link(self.obj.links, one_pub.doi)
 
-            del self.properties[PUBLICATIONS]
+            del self.properties[PUBLICATIONS_PROP]
             return
 
         # One publication and link to remove
         remove_link(self.obj.links, publication.doi)
         to_remove = publication.to_dict()
-        self.properties[PUBLICATIONS].remove(to_remove)
+        self.properties[PUBLICATIONS_PROP].remove(to_remove)
 
-        if not self.properties[PUBLICATIONS]:
-            del self.properties[PUBLICATIONS]
+        if not self.properties[PUBLICATIONS_PROP]:
+            del self.properties[PUBLICATIONS_PROP]
 
     @classmethod
     def get_schema_uri(cls) -> str:
         return SCHEMA_URI
 
     @staticmethod
     def ext(obj: T) -> "ScientificExtension[T]":
+        """Extends the given STAC Object with properties from the :stac-ext:`Scientific
+        Extension <scientific>`.
+
+        This extension can be applied to instances of :class:`~pystac.Item` or
+        :class:`~pystac.Collection`.
```

Comment:
I was curious about if there was a way to autodoc the type bounds of T, and found [this extension](https://sphinx-toolbox.readthedocs.io/en/latest/extensions/more_autodoc/typevars.html#directive-autotypevar). I tried it on this branch and it didn't look that great. Probably not worth it, but considering we removed type information from the docstrings maybe there's a way to get this from the TypeVar? If not, I think having type bounds in the docs for generic types is reasonable, so that a docs user can know what to expect.

Reply:
Yeah, I experimented a bit with [sphinx-autodoc-typehints](https://pypi.org/project/sphinx-autodoc-typehints/) as well, but that one has some pretty strict limitations on dealing with [circular imports](https://github.com/agronholm/sphinx-autodoc-typehints#dealing-with-circular-imports) that would mean we need to change the way we import a lot of our type variables. 

If we come up with a better solution using autodoc we can definitely change these down the road.